src/core/net/dns_client.hpp

```diff
@@ -108,6 +108,19 @@ class Client : public InstanceLocator, private NonCopyable
             kFlagNoRecursion      = OT_DNS_FLAG_NO_RECURSION,      ///< Server can not resolve the query recursively.
         };
 
+#if OPENTHREAD_CONFIG_DNS_CLIENT_NAT64_ENABLE
+        /**
+         * This enumeration type represents the NAT64 mode.
+         *
+         */
+        enum Nat64Mode
+        {
+            kNat64Unspecified = OT_DNS_NAT64_UNSPECIFIED, ///< NAT64 mode is not specified.
```

Comment:
```suggestion
            kNat64Unspecified = OT_DNS_NAT64_UNSPECIFIED, ///< NAT64 mode is not specified. Use default NAT64 mode.
```

Reply:
> I guess the DNS-SD server needs to implement additional A RR for AAAA query.

Client will query for `A` in a new request/message if it is not included in the original `AAAA` query response.
So we don't require the server to include it. It is optional but it will be a good optimization.

Reply:
Should we include "Use default NAT64 mode." in the comment, as @simonlingoogle  suggested aove?

Reply:
I think I may have missed adding it. Thanks.

Reply:
Looks like it was already in latest commit (but somehow it does not show here that its "outdated").

Reply:
I see! I added it in `dns_clinet.h` but still missing in `.hpp` one.